infra/gcp/bash/ensure-prod-storage.sh

```diff
@@ -250,13 +251,28 @@ function ensure_all_prod_projects() {
 
 # Create all prod GCS buckets.
 function ensure_all_prod_buckets() {
+    local sa_name="k8s-infra-push-bot"
+    local sa_email="${sa_name}@${PROD_PROJECT}.iam.gserviceaccount.com"
+    local principal="serviceAccount:${sa_email}"
+
+    color 6 "Ensuring ${sa_email} exists and can write to prod buckets in project: ${PROD_PROJECT}"
+    ensure_service_account \
+      "${PROD_PROJECT}" \
+      "${sa_name}" \
+      "used by automation to push artifacts to prod buckets in ${PROD_PROJECT}"
```

Comment:
@ameukam @xmudrii is there any automation required to get the service account private keys? I assume we manually create them and push them into the 1Password vault, right?

Reply:
We prevent private keys creation. Workload Identity or Service Account impersonation are the only things allowed now.

Reply:
Switched to workload identity as described in https://github.com/google-github-actions/auth